src/Symfony/Component/Security/Http/Firewall/SwitchUserListener.php

```diff
@@ -158,7 +158,7 @@ private function attemptSwitchUser(Request $request)
      */
     private function attemptExitUser(Request $request)
     {
-        if (false === $original = $this->getOriginalToken($this->tokenStorage->getToken())) {
+        if (null === ($currentToken = $this->tokenStorage->getToken()) || false === $original = $this->getOriginalToken($currentToken)) {
```

Comment:
What about returning `null` instead of `false` in `getOriginalToken()` so that we need only one check here?